### ISSUE TITLE
Rule sets are not removed from listener when rule_set_names is.

### ISSUE THREAD

**Issue by stanicek** [terraform-providers/terraform-provider-oci#727]
### Terraform Version

v0.11.11

### OCI Provider Version

3.17.0

### Description:
Repro steps:
1. Create a LB with a rule set attached to a listener.
```
resource "oci_load_balancer_load_balancer" "load_balancer" {
  compartment_id = "${var.compartment_ocid}"
  display_name   = "${var.load_balancer_name}"
  is_private     = "false"
  shape          = "${var.lbaas_shape_100mbps}"
  subnet_ids     = ["${oci_core_subnet.lbaas-ad1.id}", "${oci_core_subnet.lbaas-ad2.id}"]
}

resource "oci_load_balancer_rule_set" "default_rule_set" {
  name = "default"
  load_balancer_id = "${oci_load_balancer_load_balancer.load_balancer.id}"
  items = [
  {
    action = "ADD_HTTP_REQUEST_HEADER"
    header = "X-LBaaS-Test"
    value = "Test"
  },
  ]
}
resource "oci_load_balancer_listener" "default_http_listener" {
  load_balancer_id         = "${oci_load_balancer_load_balancer.load_balancer.id}"
  name                     = "default_http"
  default_backend_set_name = "${oci_load_balancer_backend_set.default_backend_set.name}"
  rule_set_names = ["${oci_load_balancer_rule_set.default_rule_set.name}"]
  port                     = "80"
  protocol                 = "HTTP"
}
```
2. Remove rule_set_names parameter from the listener.
```
resource "oci_load_balancer_listener" "default_http_listener" {
  load_balancer_id         = "${oci_load_balancer_load_balancer.load_balancer.id}"
  name                     = "default_http"
  default_backend_set_name = "${oci_load_balancer_backend_set.default_backend_set.name}"
  port                     = "80"
  protocol                 = "HTTP"
}
```

Expected:
Rule set will be detached from the listener.

Actual:
Nothing happens (empty plan).

### Terraform Plan
```
Apply complete! Resources: 0 added, 0 changed, 0 destroyed.

Outputs:
```

**Comment by kilokahn** [comment]
@stanicek this seems like a genuine bug, and we will address the fix along with #728

**Comment by codycushing** [comment]
Hello, removing a field will not trigger a change in resource state. In this case you need to set 
_rule_set_names_ to an empty array and apply

`rule_set_names = []`
